File: framework/languages/java/e2e/SpotService/Client/src/main/java/systems/zlink/e2e/spotservice/client/Scenarios/SmA3Scenario.java
package systems.zlink.e2e.spotservice.client.Scenarios;

import systems.zlink.e2e.spotservice.shared.Contracts;

public final class SmA3Scenario extends SpotServiceScenarioContext {
    private SmA3Scenario(SpotServiceScenarioContext context) {
        super(context);
    }

    public static void run(SpotServiceScenarioContext context) {
        new SmA3Scenario(context).execute();
    }

    private void execute() {
        Contracts.StateRes roomA = eventually(() -> requestState("room-a", "owner-a", REQUEST_TIMEOUT));
        Contracts.StateRes roomB = eventually(() -> requestState("room-b", "owner-b", REQUEST_TIMEOUT));
        ensure("play-a".equals(roomA.nodeRid()), "SM-A3 room-a owner mismatch");
        ensure("play-b".equals(roomB.nodeRid()), "SM-A3 room-b owner mismatch");
        System.out.println("scenario SM-A3 passed");

    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Client/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/client/scenarios/SmA3Scenario.kt
package systems.zlink.e2e.kotlin.spotservice.client.scenarios

import systems.zlink.e2e.kotlin.spotservice.client.support.ensure
import systems.zlink.e2e.kotlin.spotservice.client.support.eventually
import systems.zlink.e2e.kotlin.spotservice.client.support.SpotHttpDriver

internal object SmA3Scenario {
    suspend fun run(spots: SpotHttpDriver) {
        val roomA = eventually {
            spots.requestState("room-a", "owner-a")
        }
        ensure(roomA.nodeRid == "play-a", "SM-A3 room-a owner mismatch")
        println("scenario SM-A3 passed")
    }
}
